packages/expo-ui/android/src/main/java/expo/modules/ui/SingleChoiceSegmentedControlView.kt

```diff
@@ -0,0 +1,46 @@
+package expo.modules.ui
+
+import android.content.Context
+import androidx.compose.material3.SegmentedButton
+import androidx.compose.material3.SegmentedButtonDefaults
+import expo.modules.kotlin.viewevent.EventDispatcher
+import expo.modules.kotlin.views.ExpoComposeView
+import androidx.compose.material3.SingleChoiceSegmentedButtonRow
+import androidx.compose.material3.Text
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import expo.modules.kotlin.AppContext
+
+data class SegmentedControlProps(
+  val options: MutableState<Array<String>> = mutableStateOf(emptyArray()),
+  val selectedIndex: MutableState<Int?> = mutableStateOf(null)
+)
+
+class SingleChoiceSegmentedControlView(context: Context, appContext: AppContext) : ExpoComposeView(context, appContext) {
+  override val props: SegmentedControlProps = SegmentedControlProps()
+  private val onOptionSelected by EventDispatcher()
+
+  init {
+    setContent {
```

Comment:
i tried to checkout this branch and tested locally. there was an exception for me and no SingleChoiceSegmentedControlView was shown:
```
java.lang.NoSuchMethodError: No virtual method setContent(Lkotlin/jvm/functions/Function2;)V in class Lexpo/modules/ui/SingleChoiceSegmentedControlView; or its super classes (declaration of 'expo.modules.ui.SingleChoiceSegmentedControlView' appears in /data/app/~~Tko5e6BP8rFgqyAT6RWiaQ==/dev.expo.payments-PpfD5sVHI5F8n4Ra9pUkNQ==/base.apk!classes2.dex)
```

maybe the kotlin infer type is incorrect here in the lambda.

Reply:
Oh no, I probably stripped something important last second when cleaning up the PR, let me fix it :)

Reply:
Fixed – sorry for that Kudo!